src/Akeneo/Pim/Enrichment/Component/Product/Job/DeleteProductsAndProductModelsTasklet.php

```diff
@@ -189,9 +172,19 @@ protected function doDelete(array $entities): void
         $products = $this->filterProducts($entities);
         $productModels = $this->filterProductModels($entities);
 
-        $deletedProductsCount = $this->countProductsToDelete($products, $productModels);
         $deletedProductModelsCount = $this->countProductModelsToDelete($productModels);
 
+        // When deletion for product models is not granted, we add a warning and reset the array so that there will be
+        // no deletion nor for their children
+        if (!$this->securityFacade->isGranted('pim_enrich_product_model_remove')) {
+            $this->stepExecution->addWarning('Access forbidden. You are not allowed to delete product models', [], new DataInvalidItem($productModels));
+            $this->stepExecution->incrementSummaryInfo('skipped_deleted_product_models', $deletedProductModelsCount);
+            $deletedProductModelsCount = 0;
+            $productModels = [];
```

Comment:
You can't early return ? It's less complex

```suggestion
            return 0;
```

Reply:
We still want to get through the following code, which handles the step execution informations and the product deletion.
But it's maybe not clear enough in the comment? (let me know)